andela/ah-frontend-athena
# #162163229 Users should be able to login via Facebook, Google

#### What does this PR do?
Enable Author's Haven users login into the application using Google or Facebook.

#### Description of tasks to be completed?
- Install Google and Facebook packages to get tokens from the corresponding platform
- Make a call to the backend API endpoint to validate/process the retrieved `access_token`
- Log the user in (on success), save returned JWT, and redirect the user to the landing (articles) page OR show error messages if the login failed.
 
#### How can this be manually tested?
Open the PR review app and click on a social login button. On the first try, this should open a pop-up browser where the social platform(s) asks you to authorize our app by logging it clicking authorize.

#### What are the relevant pivotal tracker stories?
[#162163229](https://www.pivotaltracker.com/story/show/162163229)

#### Screenshots
Facebook pop-up
![image](https://user-images.githubusercontent.com/5712103/51521888-fa6d9c00-1e38-11e9-8aaf-4648dad31af5.png)

Successful Facebook login
![image](https://user-images.githubusercontent.com/5712103/51521879-f3df2480-1e38-11e9-9733-15e7130d19f2.png)

Google pop-up
![image](https://user-images.githubusercontent.com/5712103/51522000-39035680-1e39-11e9-8c47-b59cac1a1d28.png)

Successful Google login
![image](https://user-images.githubusercontent.com/5712103/51521770-a8c51180-1e38-11e9-957e-3a2ca29171b7.png)
